### PR TITLE
Update nuheat for new climate platform

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Remove docs for pre climate 1.0 items that no longer exist

Add notes about removal of the off mode as it doesn't actually work since nuheat doesn't implement it.

Implement missing set_hvac_mode for nuheat

Fix hvac_mode as it was really implementing hvac_action

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/32714
- This PR fixes or closes issue:

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/nuheat.markdown [hunk 1]
```diff
@@ -136,28 +136,19 @@ Returns the maximum supported temperature by the thermostat
 
 The following services are provided by the NuHeat Thermostat: `set_temperature`, `set_hvac_mode`, `set_preset_mode`, `resume_program`.
 
-The services `fan_min_on_time`, `set_aux_heat`, `set_away_mode`, `set_humidity`, `set_fan_mode`, and `set_swing_mode` offered by the [Climate component](/integrations/climate/) are not implemented for this thermostat.
+### Service `climate.set_hvac_mode` ([Climate component](/integrations/climate/))
```
Comment (frenck) [inline source/_integrations/nuheat.markdown]:
```suggestion
### Service `climate.set_hvac_mode` ([Climate integration](/integrations/climate/))
```

File: source/_integrations/nuheat.markdown [hunk 2]
```diff
@@ -136,28 +136,19 @@ Returns the maximum supported temperature by the thermostat
...
 
-### Service `set_temperature`
+NuHeat Thermostats do not have an off concept. Setting the temperature to `min_temp` and changing the mode to `heat` will cause the device to enter a `Permanent Hold` preset and will stop the thermostat from turning on unless you happen to live in a freezing climate.
 
-Puts the thermostat into an indefinite hold at the given temperature.
+### Service `climate.set_temperature` ([Climate component](/integrations/climate/))
```
Comment (frenck) [inline source/_integrations/nuheat.markdown]:
```suggestion
### Service `climate.set_temperature` ([Climate integration](/integrations/climate/))
```

File: source/_integrations/nuheat.markdown [hunk 3]
```diff
@@ -136,28 +136,19 @@ Returns the maximum supported temperature by the thermostat
...
-
-Only the target temperatures relevant for the current operation mode need to
-be provided.
+If the thermostat is in auto mode, it puts the thermostat into a temporary hold at the given temperature.
 
-### Service `set_preset_mode`
+If the thermostat is in heat mode, it puts the thermostat into a permanent hold at the given temperature.
 
-Sets the thermostat's preset mode. Without a preset mode set it run the thermostat's programmed schedule, "temperature" (to indefinitely hold the thermostat's current target temperature), or "temporary_temperature" (to hold the thermostat's current target temperature until the thermostat's next scheduled event).
+### Service `climate.set_preset_mode` ([Climate component](/integrations/climate/))
```
Comment (frenck) [inline source/_integrations/nuheat.markdown]:
```suggestion
### Service `climate.set_preset_mode` ([Climate integration](/integrations/climate/))
```

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks, @bdraco. 3 Very small suggestions left on a review.

Could you take a look? Thanks 👍

File: source/_integrations/nuheat.markdown (hunk 3 above)
Comment (bdraco) [reply source/_integrations/nuheat.markdown]: Thanks. This one has seen a lot of love in a long time :)

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Pull request merged**